### PR TITLE
Parallel benchmark

### PR DESCRIPTION
**Pull request by jakobbossek**
* reworked `cocoSuiteGetAllproblems`: We cannot use `coco_suite_get_next_problem` here, because the problem instance returned by this function is available only up to the next call.
* added `cocoInitObserver` to initialize an observer
* added `cocoProblemAddObserver` to wrap a problem with an observer
* added `cocoProblemFree` to free memory of a given problem (this is needed for parallelization)
* `cocoBenchmarkOptimizer` now makes use of parallelMap
* `cocoInitSuite` now does not initialize the bbob observer transparently. This is done explicitly via `cocoInitObserver`

### CODE REVIEW

File: R/cocoBenchmarkOptimizer.R [hunk 1]
```diff
@@ -7,28 +7,43 @@
...
+  assertClass(observer, "CocoObserver")
+
+  problems = cocoSuiteGetAllProblems(suite)
+  problem.ids = names(problems)
+
+  res = parallelMap(function(id) {
+    p = problems[[id]]
+    # wrap function with observer
+    p = cocoProblemAddObserver(p, observer)
+    cocoRunOptimizer(cocoWrapperOptimNelderMead, p, ...)
```
Comment (berndbischl) [inline R/cocoBenchmarkOptimizer.R]: bad bug? why nelder mead here?

File: R/cocoProblemInitObserver.R [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#' @title Init coco observer.
+#'
+#' @description
+#' See title
```
Comment (berndbischl) [inline R/cocoProblemInitObserver.R]: i would add 2 lines here what an observer actually does and is for. 
also a space before next at_param

File: R/cocoProblemFree.R [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#' @title Free the given coco problem
```
Comment (berndbischl) [inline R/cocoProblemFree.R]: dot after title.

File: R/cocoProblemFree.R [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+#'
+#' @description
+#' See title
```
Comment (berndbischl) [inline R/cocoProblemFree.R]: tell the user when this should be called and what it does

File: man-roxygen/arg_observer.R [hunk 5]
```diff
@@ -0,0 +1,2 @@
+#' @param observer [\code{character(1)}]\cr
```
Comment (berndbischl) [inline man-roxygen/arg_observer.R]: actually the docs are wrong? observer is of class "CocoObserver" isnt it?

Comment (berndbischl) [inline man-roxygen/arg_observer.R]: and what the Oberserver does should be doced in initCocoObserver, which is linked anyhow when the class is correctly mentioned here...

**Comment by berndbischl** [comment]
this is REALLY great.

i have added some more less minor comments. i would suggest to look at these, when they are fixed we should probably just mess around with the current functionalityl and see what works - and what is problematic

File: R/cocoBenchmarkOptimizer.R (hunk 1 above)
Comment (jakobbossek) [inline R/cocoBenchmarkOptimizer.R]: OMG!

File: R/cocoProblemFree.R (hunk 3 above)
Comment (jakobbossek) [inline R/cocoProblemFree.R]: Done

File: R/cocoProblemInitObserver.R (hunk 2 above)
Comment (jakobbossek) [inline R/cocoProblemInitObserver.R]: Done

File: man-roxygen/arg_observer.R (hunk 5 above)
Comment (jakobbossek) [inline man-roxygen/arg_observer.R]: Actually this is wrong. An observer is of S3 class CocoObserver.

File: R/cocoProblemFree.R (hunk 4 above)
Comment (jakobbossek) [inline R/cocoProblemFree.R]: 👍

File: man-roxygen/arg_observer.R (hunk 5 above)
Comment (jakobbossek) [inline man-roxygen/arg_observer.R]: Done

**Comment by jakobbossek** [comment]
Ok. Great! Hopefully fixed all of your comments.

**Comment by berndbischl** [comment]
i will just merge this now. 

we can then easily work with the master, and if stuff is not perfect create more smaller issues

**Pull request merged**